v2/pkg/genruntime/indexer.go

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genruntime
+
+type Indexer interface {
```

Comment:
Should we consider having `Index` from `v2/pkg/genruntime/registration/registration.go` and this at the same place?

Reply:
Possibly. `Indexer` is a bit more generic than `Index`, as `Index` is really used specifically for our registration, whereas in theory `Indexer` could be used generically to define any index, so I do think it's probably ok that one of them is in `genruntime` while one is in `genruntime/registration`, reflecting its slightly more scoped nature.

In practice right now they're both basically used in the same places, but that isn't necessarily going to be the case forever.